gobblin-data-management/src/main/java/gobblin/data/management/copy/CopySource.java

```diff
@@ -187,4 +198,21 @@ public static void serializeCopyableDataset(State state, CopyableDatasetMetadata
   public static CopyableDatasetMetadata deserializeCopyableDataset(State state) throws IOException {
     return CopyableDatasetMetadata.deserialize(state.getProp(SERIALIZED_COPYABLE_DATASET));
   }
+
+  private Collection<Partition<CopyableFile>> partitionCopyableFiles(Collection<CopyableFile> files) {
+    Map<String, Partition.Builder<CopyableFile>> partitionBuildersMaps = Maps.newHashMap();
+    for (CopyableFile file : files) {
+      if (!partitionBuildersMaps.containsKey(file.getPartition())) {
+        partitionBuildersMaps.put(file.getPartition(), new Partition.Builder<CopyableFile>(file.getPartition()));
+      }
+      partitionBuildersMaps.get(file.getPartition()).add(file);
+    }
+    return Lists.newArrayList(Iterables.transform(partitionBuildersMaps.values(),
```

Comment:
I think the result is that it looks like Perl -- a sequence of seemingly random characters. :-)